src/GraphQL.Tests/Errors/ErrorParserTests.cs

```diff
@@ -0,0 +1,243 @@
+        {
+            var error = new ExecutionError(null) { Code = "test code" };
+
+            var parsed = new ErrorParser().Parse(error);
+            parsed.Message.ShouldBe(error.Message);
+            parsed.Locations.ShouldBeNull();
+            parsed.Path.ShouldBeNull();
+            parsed.Extensions.ShouldNotBeNull();
+            parsed.Extensions.Count.ShouldBe(2);
+            parsed.Extensions.ShouldContainKeyAndValue("code", "test code");
+            parsed.Extensions.ShouldContainKey("codes");
+            parsed.Extensions["codes"].ShouldBeAssignableTo<IEnumerable<string>>().ShouldBe(new[] { "test code" });
+        }
+
+        [Fact]
+        public void message_and_inner_exception()
+        {
+            var innerException = new ArgumentNullException(null, new ArgumentOutOfRangeException());
+            var error = new ExecutionError(innerException.Message, innerException);
+            //error.Code.ShouldBe("ARGUMENT_NULL");
+            error.Code.ShouldNotBeNull();
+            //error.Codes.ShouldBe(new[] { "ARGUMENT_NULL", "ARGUMENT_OUT_OF_RANGE" });
+            error.Codes.Count().ShouldBe(2);
+
+            var parsed = new ErrorParser().Parse(error);
+            parsed.Message.ShouldBe(error.Message);
+            parsed.Locations.ShouldBeNull();
+            parsed.Path.ShouldBeNull();
+            parsed.Extensions.ShouldNotBeNull();
+            parsed.Extensions.Count.ShouldBe(2);
+            parsed.Extensions.ShouldContainKeyAndValue("code", error.Code);
+            parsed.Extensions.ShouldContainKey("codes");
+            parsed.Extensions["codes"].ShouldBeAssignableTo<IEnumerable<string>>().ShouldBe(error.Codes);
+        }
+
+        [Fact]
+        public void message_and_single_location()
+        {
+            var error = new ExecutionError(null);
+            error.AddLocation(1, 2);
+            error.Locations.Count().ShouldBe(1);
+
+            var parsed = new ErrorParser().Parse(error);
+            parsed.Message.ShouldBe(error.Message);
+            parsed.Locations.ShouldBe(error.Locations);
+            parsed.Path.ShouldBeNull();
+            parsed.Extensions.ShouldBeNull();
+        }
+
+        [Fact]
+        public void message_and_multiple_locations()
+        {
+            var error = new ExecutionError(null);
+            error.AddLocation(1, 2);
+            error.AddLocation(3, 4);
+            error.Locations.Count().ShouldBe(2);
+
+            var parsed = new ErrorParser().Parse(error);
+            parsed.Message.ShouldBe(error.Message);
+            parsed.Locations.ShouldBe(error.Locations);
+            parsed.Path.ShouldBeNull();
+            parsed.Extensions.ShouldBeNull();
+        }
+
+        [Fact]
+        public void message_and_path()
+        {
+            var error = new ExecutionError(null)
+            {
+                //Path = new object[] { "root", 23, "child" },
+                Path = new string[] { "root", "23", "child" },
```

Comment:
```suggestion
                Path = new object[] { "root", 23, "child" },
```
Once #1657 is pulled in, we can update this test to appropriately reflect the sample.